man/tpm2_policyauthorize.1.md

```diff
@@ -0,0 +1,85 @@
+% tpm2_policyauthorize(1) tpm2-tools | General Commands Manual
+%
+% AUGUST 2018
+
+# NAME
+
+**tpm2_policyauthorize**(1) - Generates/ Creates a policy event that authorizes 
+a policy digest from TPM policy events.
+
+# SYNOPSIS
+
+**tpm2_policyauthorize** [*OPTIONS*]
+
+# DESCRIPTION
+
+**tpm2_policyauthorize**(1) generates a policy Authorize event with the TPM. It 
+expects a session to be already established via **tpm2_startauthsession**(1). If
+the input session is a trial session this tool generates a policy digest that 
+associates a signing authority's public key name with the policy being 
+authorized to run. If the input session is a policy session tpm2_policyauthorize
+looks for a verification ticket from the TPM to attest that the TPM has verified
+the signature on the policy digest.
+
+# OPTIONS
+
+  * **-f**, **--policy-file**=_POLICY\_FILE_:
+    File to save the policy digest.
+
+  * **-S**, **--session**=_SESSION_FILE_:
+
+    The policy session file generated via the **-S** option to
+    **tpm2_startauthsession**(1).
+
+  * **-a**, **--input-policy-file**=_POLICY\_FILE_:
+
+    The policy digest that has to be authorized.
+
+  * **-b**, **--input-policy-qualifier-file**=_DATA_FILE_:
+
+    The policy qualifier data signed in conjunction with the input policy digest.
+    This is a unique data that the signer can choose to include in the signature.
```

Comment:
Is this really mandatory? Can they leave it blank? Can we just call openssl_rand() internally and get this if not specified? If we generate does the user actually need this value?

Reply:
@williamcroberts you are absolutely right. This is not mandatory. However this cannot be random either. It's a known fixed data alongside policy digest that has to be signed. Making this an optional input, in absence of which, it will be a blank TPM2B_NONCE data.